pkg/services/sso/keycloak_service_proxy.go

```diff
@@ -26,6 +27,29 @@ func (r *keycloakServiceProxy) retrieveToken() (string, *errors.ServiceError) {
 	return accessToken, nil
 }
 
+func retrieveUserToken(ctx context.Context) (string, *errors.ServiceError) {
+	userToken, err := authentication.TokenFromContext(ctx)
+	if err != nil {
+		return "", errors.NewWithCause(errors.ErrorGeneral, err, "error getting access token")
+	}
+	token := userToken.Raw
+	return token, nil
+}
+
+func tokenForServiceAPIHandler(ctx context.Context, r *keycloakServiceProxy) (string, *errors.ServiceError) {
+	var token string
+	var err *errors.ServiceError
+	if r.GetConfig().SelectSSOProvider == keycloak.REDHAT_SSO {
+		token, err = retrieveUserToken(ctx)
+	} else {
+		token, err = r.retrieveToken()
+	}
+	if err != nil {
+		return "", err
+	}
+	return token, nil
+}
+
```

Comment:
The `tokenProvider` interface should be used to provide the token.
Both RedhatSSO Service implementation and Keycloak Service implementation should provide a tokenProvider that returns the token as needed.

This way we will avoid the `if r.GetConfig().SelectSSOProvider == keycloak.REDHAT_SSO` and it will be more extensible/maintainable in the event we will need to add/remove an implementation in the future.

Reply:
Agreed, but this is a different type of token. the "user" token needs to be retrieved from the request header and pass it to sso api.